src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaTransportPump.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+using CoreWCF.Configuration;
+using CoreWCF.Queue.Common;
+using CoreWCF.Runtime;
+using Microsoft.Extensions.Logging;
+
+namespace CoreWCF.Channels;
+
+internal sealed class KafkaTransportPump : QueueTransportPump, IDisposable
+{
+    private readonly ILogger<KafkaTransportPump> _logger;
+    private readonly KafkaDeliverySemantics _kafkaDeliverySemantics;
+    internal IConsumer<Null, byte[]> Consumer { get; private set; }
+    internal ConsumerConfig ConsumerConfig { get; private set; }
+    internal IProducer<Null, byte[]> Producer { get; private set; }
+    internal string Topic { get; }
+    internal KafkaTransportBindingElement TransportBindingElement { get; }
+    private CountdownEvent _receiveContextCountdownEvent;
+
+    private readonly Uri _baseAddress;
+    private CancellationTokenSource _cts;
+    private AsyncManualResetEvent _mres;
+    private static readonly (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) s_atMostOnceConfigValues = (false, null);
+    private static readonly (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) s_atLeastOncePerMessageCommitConfigValues = (false, null);
+    private static readonly (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) s_atLeastOnceBatchCommitConfigValues = (true, false);
+    private static readonly Regex s_topicNameRegex =
+        new(@"^[a-zA-Z0-9\.\-_]{1,255}$", RegexOptions.Compiled, TimeSpan.FromMilliseconds(100));
```

Comment:
This can be converted to a source generated regex. See [https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators) for more info.

Reply:
CoreWCF.Kafka is targeting netstandard2.0 but it looks like this feature is only available starting from dotnet7.